### PR TITLE
OADP-2130: Restructure of the OADP Introduction

### PR DESCRIPTION
**Pull request by anarnold97**
[OADP-2130](https://issues.redhat.com/browse/OADP-2130)

Doc Improvements - Restructure of OADP documentation

This PR is a focused on including an introduction section and improving the structure of the documentation as the taxonomy was problematic. 

Enterprise 4.11 → branch/enterprise-4.11
Enterprise 4.12 → branch/enterprise-4.12
Enterprise 4.13 → branch/enterprise-4.13
Enterprise 4.14 → branch/enterprise-4.14
Enterprise 4.15 → branch/enterprise-4.15

Issue:

https://issues.redhat.com/browse/OADP-2130

Link to docs preview:

[OADP Introduction](https://64198--docspreview.netlify.app/openshift-enterprise/latest/backup_and_restore/application_backup_and_restore/oadp-intro)

QE review:
- [ X] QE has approved this change.

Additional information:

### CODE REVIEW

**Comment by weshayutin** [comment]
LGTM, only comment would be missing links to the following.  Not a blocker for this PR though imho
```
BackupStorageLocation

VolumeSnapshotLocation
```

File: backup_and_restore/application_backup_and_restore/oadp-intro.adoc
Comment (stillalearner) [inline backup_and_restore/application_backup_and_restore/oadp-intro.adoc:23]: this link should redirect to the Scheduling Backups section.
https://64198--docspreview.netlify.app/openshift-enterprise/latest/backup_and_restore/application_backup_and_restore/oadp-api#oadp-api

I dont see anything related to Schedule on the current link it redirects to.

Comment (anarnold97) [reply backup_and_restore/application_backup_and_restore/oadp-intro.adoc:23]: No

Comment (anarnold97) [reply backup_and_restore/application_backup_and_restore/oadp-intro.adoc]: @weshayutin && @akarol && @stillalearner - let me explain - I am working mostly on structural changes, once I have the structure changed I think it will be easier to add re-writes. 
Please can I just get a LGTM and I can get this moving

**Review by stillalearner — APPROVED** [review]
lgtm

**Comment by akarol** [comment]
LGTM. approved

File: backup_and_restore/application_backup_and_restore/oadp-intro.adoc
Comment (stevsmit) [inline backup_and_restore/application_backup_and_restore/oadp-intro.adoc:3]: We should reorganize this metadata info so that it follows OCP guidelines: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#assembly-file-metadata

```suggestion
:_content-type: ASSEMBLY
[id="oadp-introduction"]
= Introduction to {oadp-full}
include::_attributes/common-attributes.adoc[]
:context: oadp-api
:namespace: openshift-adp
:local-product: OADP

toc::[]
```

File: backup_and_restore/application_backup_and_restore/oadp-intro.adoc
Comment (stevsmit) [inline backup_and_restore/application_backup_and_restore/oadp-intro.adoc:15]: ". . . information on" is used, but ". . . information about" is more commonly used in our docs:

I have a few suggestions here for how better to word this for clarity, because I don't think the link necessarily provides information about "etcd", but rather "backing up etcd".

1:
```suggestion
OADP cannot be used as a disaster recovery solution for etcd or OpenShift Operators. See the documentation for xref:../../backup_and_restore/control_plane_backup_and_restore/backing-up-etcd.adoc#backup-etcd[Backing up etcd] for more information. 
```
2:
```suggestion
For more information about backing up etcd, see xref:../../backup_and_restore/control_plane_backup_and_restore/backing-up-etcd.adoc#backup-etcd[Backing up etcd].
```

That said, I'm not really crazy about including the link to this information right here. I think that it could go on the bottom of the page, after the content, in an additional resources section. See comment below for example. 

Kind of up to you how you want to proceed.

File: backup_and_restore/application_backup_and_restore/oadp-intro.adoc
Comment (stevsmit) [inline backup_and_restore/application_backup_and_restore/oadp-intro.adoc:26]:
```suggestion
[role="_additional-resources"]
.Additional resources

* xref:../../backup_and_restore/control_plane_backup_and_restore/backing-up-etcd.adoc#backup-etcd[Backing up etcd]
```

File: backup_and_restore/application_backup_and_restore/oadp-intro.adoc
Comment (stevsmit) [inline backup_and_restore/application_backup_and_restore/oadp-intro.adoc:11]: Gonna take a stab and rewriting/combining these first two/three paragraphs: 

```suggestion
The {oadp-first} product safeguards customer applications on {product-title}. It offers comprehensive disaster recovery protection, covering OpenShift Container Platform applications, application-related cluster resources, persistent volumes, and internal images. OADP is also capable of backing up both containerized applications and virtual machines (VMs). However, it does not serve as a disaster recovery solution for etcd or OpenShift Operators. 
```

File: backup_and_restore/application_backup_and_restore/oadp-intro.adoc
Comment (stevsmit) [inline backup_and_restore/application_backup_and_restore/oadp-intro.adoc:16]: I'm kind of playing around with the idea of combining those three paragraphs (as I did above), and then making a separate section for the APIs.  eg:

```suggestion
// Module included in the following assemblies:
//
// * backup_and_restore/application_backup_and_restore/oadp-intro.adoc

[id="oadp-apis_{context}"]
= {oadp-full} APIs

{oadp-first} provides APIs to enable multiple approaches to customize backups and prevent the inclusion of unnecessary or inappropriate resources. 

OADP provides the following APIs:

* xref:../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/backing-up-applications.adoc#backing-up-applications[Backup]
* xref:../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/restoring-applications.adoc#restoring-applications[Restore]
* Schedule
* BackupStorageLocation
* VolumeSnapshotLocation


```

**Review by stevsmit — COMMENTED** [review]
Mostly considerations to reorganize the content. The metadata content should be addressed, but I didn't see any glaring issues according to IBMSG. I think the reorg might look nice. 

Good job.

File: backup_and_restore/application_backup_and_restore/oadp-intro.adoc
Comment (anarnold97) [reply backup_and_restore/application_backup_and_restore/oadp-intro.adoc]: Thanks for the re-work

I would like to keep the mention of etc highlighted, using whitespace. So I am going to drop that line down.

File: backup_and_restore/application_backup_and_restore/oadp-intro.adoc
Comment (mburke5678) [inline backup_and_restore/application_backup_and_restore/oadp-intro.adoc:19]:
```suggestion
The {oadp-first} product safeguards customer applications on {product-title}. It offers comprehensive disaster recovery protection, covering {product-title} applications, application-related cluster resources, persistent volumes, and internal images. OADP is also capable of backing up both containerized applications and virtual machines (VMs). 
```

File: backup_and_restore/application_backup_and_restore/oadp-intro.adoc
Comment (mburke5678) [inline backup_and_restore/application_backup_and_restore/oadp-intro.adoc:27]:
```suggestion
{oadp-first} provides APIs that enable multiple approaches to customizing backups and preventing the inclusion of unnecessary or inappropriate resources. 
```

File: backup_and_restore/application_backup_and_restore/oadp-intro.adoc
Comment (mburke5678) [inline backup_and_restore/application_backup_and_restore/oadp-intro.adoc:16]: Why is the [title and metadata repeated](https://github.com/openshift/openshift-docs/pull/64198/files#diff-1ee50b000b47077286254c47ff98fdc2505e4a0b5f2b7c2d4b464d8816860ccfR5-R15)?

File: backup_and_restore/application_backup_and_restore/oadp-intro.adoc
Comment (mburke5678) [inline backup_and_restore/application_backup_and_restore/oadp-intro.adoc:46]: Why is [this text repeated](https://github.com/openshift/openshift-docs/pull/64198/files#diff-1ee50b000b47077286254c47ff98fdc2505e4a0b5f2b7c2d4b464d8816860ccfR40-R46)?

File: backup_and_restore/application_backup_and_restore/oadp-intro.adoc
Comment (mburke5678) [inline backup_and_restore/application_backup_and_restore/oadp-intro.adoc:47]: Need a carrieage return before the `[role="_additional-resources"]`.

**Comment by mburke5678** [comment]
@anarnold97 A few issues that should be addressed before merging.

File: backup_and_restore/application_backup_and_restore/oadp-intro.adoc
Comment (anarnold97) [reply backup_and_restore/application_backup_and_restore/oadp-intro.adoc:29]: ta muchly

File: backup_and_restore/application_backup_and_restore/oadp-intro.adoc
Comment (anarnold97) [reply backup_and_restore/application_backup_and_restore/oadp-intro.adoc]: Cos I made a mistake

**Comment by anarnold97** [comment]
@mburke5678 - thanks very much for your time.
I made mistakes when working late, for which I apologize.

**Pull request merged**
